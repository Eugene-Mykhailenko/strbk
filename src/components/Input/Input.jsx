/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-access-state-in-setstate */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';
import classNames from 'classnames';
import styles from './Input.scss';
import animationStyles from '../../assets/styles/base/animations.scss';
import Icon from '../Icon/Icon';

class Input extends Component {
  constructor(props) {
    super(props);

    // TODO: decide if we need dynamic type update.
    this.state = {
      type: props.type,
      isPasswordType: props.type === 'password',
      isPasswordVisible: false,
      hasFocus: false,
    };

    this.internalOnChange = this.internalOnChange.bind(this);
    this.onFocus = this.onFocus.bind(this);
    this.onBlur = this.onBlur.bind(this);
    this.togglePassword = this.togglePassword.bind(this);
  }

  onFocus() {
    this.setState({
      hasFocus: true,
    });
  }

  onBlur(event) {
    if (!event.target.value.length) {
      this.setState({
        hasFocus: false,
      });
    }
  }

  getContainerClassNames() {
    const { hasError, isRequired, disabled } = this.props;
    const { hasFocus, isPasswordType } = this.state;

    return classNames(styles['form-group'], {
      [styles['has-focus']]: hasFocus,
      [styles['is-required']]: isRequired,
      [styles['is-password-type']]: isPasswordType,
      [styles['has-error']]: hasError,
      [styles['is-disabled']]: disabled,
    });
  }

  getInputClassNames() {
    const { hasError } = this.props;
    const { hasFocus } = this.state;

    return classNames(styles['form-control'], {
      [styles['has-focus']]: hasFocus,
      [styles['has-error']]: hasError,
    });
  }

  togglePassword() {
    const isPasswordVisible = !this.state.isPasswordVisible;

    this.setState({
      isPasswordVisible,
      type: isPasswordVisible ? 'text' : 'password',
    });
  }

  internalOnChange(event) {
    const { onChange, disabled } = this.props;

    if (!disabled) {
      onChange(event);
    }
  }

  renderLabel() {
    const { placeholder } = this.props;

    return placeholder && <label className={styles.label}>{placeholder}</label>;
  }

  renderButton() {
    const { isPasswordVisible, isPasswordType } = this.state;

    return (
      isPasswordType && (
        <button className={styles.btn} onClick={this.togglePassword}>
          <Icon
            className={styles.icon}
            width={24}
            height={24}
            glyph={isPasswordVisible ? 'UII_Eye' : 'UII_EyeSlash'}
          />
        </button>
      )
    );
  }

  renderErrorText() {
    const { hasError, errorText } = this.props;

    return (
      <CSSTransition
        in={hasError}
        classNames={{
          enter: animationStyles['fade-in-start'],
          enterActive: animationStyles['fade-in-end'],
          exit: animationStyles['fade-out-start'],
          exitActive: animationStyles['fade-out-end'],
        }}
        timeout={300}
        unmountOnExit
      >
        <div className={styles['error-text']}>{errorText}</div>
      </CSSTransition>
    );
  }

  render() {
    const { disabled } = this.props;
    const { type } = this.state;

    return (
      <div className={this.getContainerClassNames()}>
        {this.renderLabel()}
        <input
          className={this.getInputClassNames()}
          type={type}
          disabled={disabled}
          onChange={this.internalOnChange}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
        />
        {this.renderButton()}
        {this.renderErrorText()}
      </div>
    );
  }
}

Input.propTypes = {
  /** Input Placeholder/Label */
  placeholder: PropTypes.string,
  /** Input field type: text, number, password and etc */
  type: PropTypes.string,
  /** Callback function for event */
  onChange: PropTypes.func.isRequired,
  /** Add star for label if this field is required */
  isRequired: PropTypes.bool,
  /** Error indication */
  hasError: PropTypes.bool,
  /** Error message */
  errorText: PropTypes.string,
  /** Disabled input */
  disabled: PropTypes.bool,
};

Input.defaultProps = {
  placeholder: undefined,
  type: 'text',
  isRequired: false,
  hasError: false,
  errorText: undefined,
  disabled: false,
};

export default Input;
