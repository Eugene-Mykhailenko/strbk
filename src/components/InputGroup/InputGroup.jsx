import React from 'react';
import PropTypes from 'prop-types';
import styles from '../Input/Input.scss';

const InputGroup = (props) => {
  const { children } = props;

  return <div className={styles['input-group']}>{children}</div>;
};

InputGroup.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default InputGroup;
