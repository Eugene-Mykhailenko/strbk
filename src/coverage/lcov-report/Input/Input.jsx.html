<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Input/Input.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Input</a> Input.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.1% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>27/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>7/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.82% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.1% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>27/31</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-access-state-in-setstate */
&nbsp;
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';
import classNames from 'classnames';
import styles from './Input.scss';
import animationStyles from '../../assets/styles/base/animations.scss';
import Icon from '../Icon/Icon';
&nbsp;
class Input extends Component {
  constructor(props) {
    super(props);
&nbsp;
    // TODO: decide if we need dynamic type update.
    this.state = {
      type: props.type,
      isPasswordType: props.type === 'password',
      isPasswordVisible: false,
      hasFocus: false,
    };
&nbsp;
    this.internalOnChange = this.internalOnChange.bind(this);
    this.onFocus = this.onFocus.bind(this);
    this.onBlur = this.onBlur.bind(this);
    this.togglePassword = this.togglePassword.bind(this);
  }
&nbsp;
  onFocus() {
    this.setState({
      hasFocus: true,
    });
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>Blur(event) {
<span class="cstat-no" title="statement not covered" >    if (!event.target.value.length) {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({</span>
        hasFocus: false,
      });
    }
  }
&nbsp;
  getContainerClassNames() {
    const { hasError, isRequired, disabled } = this.props;
    const { hasFocus, isPasswordType } = this.state;
&nbsp;
    return classNames(styles['form-group'], {
      [styles['has-focus']]: hasFocus,
      [styles['is-required']]: isRequired,
      [styles['is-password-type']]: isPasswordType,
      [styles['has-error']]: hasError,
      [styles['is-disabled']]: disabled,
    });
  }
&nbsp;
  getInputClassNames() {
    const { hasError } = this.props;
    const { hasFocus } = this.state;
&nbsp;
    return classNames(styles['form-control'], {
      [styles['has-focus']]: hasFocus,
      [styles['has-error']]: hasError,
    });
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  to</span>gglePassword() {
    const isPasswordVisible = <span class="cstat-no" title="statement not covered" >!this.state.isPasswordVisible;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      isPasswordVisible,
      type: isPasswordVisible ? 'text' : 'password',
    });
  }
&nbsp;
  internalOnChange(event) {
    const { onChange, disabled } = this.props;
&nbsp;
    if (!disabled) {
      onChange(event);
    }
  }
&nbsp;
  renderLabel() {
    const { placeholder } = this.props;
&nbsp;
    return placeholder &amp;&amp; &lt;label className={styles.label}&gt;{placeholder}&lt;/label&gt;;
  }
&nbsp;
  renderButton() {
    const { isPasswordVisible, isPasswordType } = this.state;
&nbsp;
    return (
      isPasswordType &amp;&amp; (
        &lt;button className={styles.btn} onClick={this.togglePassword}&gt;
          &lt;Icon
            className={styles.icon}
            width={24}
            height={24}
            glyph={isPasswordVisible ? <span class="branch-0 cbranch-no" title="branch not covered" >'UII_Eye' </span>: 'UII_EyeSlash'}
          /&gt;
        &lt;/button&gt;
      )
    );
  }
&nbsp;
  renderErrorText() {
    const { hasError, errorText } = this.props;
&nbsp;
    return (
      &lt;CSSTransition
        in={hasError}
        classNames={{
          enter: animationStyles['fade-in-start'],
          enterActive: animationStyles['fade-in-end'],
          exit: animationStyles['fade-out-start'],
          exitActive: animationStyles['fade-out-end'],
        }}
        timeout={300}
        unmountOnExit
      &gt;
        &lt;div className={styles['error-text']}&gt;{errorText}&lt;/div&gt;
      &lt;/CSSTransition&gt;
    );
  }
&nbsp;
  render() {
    const { disabled } = this.props;
    const { type } = this.state;
&nbsp;
    return (
      &lt;div className={this.getContainerClassNames()}&gt;
        {this.renderLabel()}
        &lt;input
          className={this.getInputClassNames()}
          type={type}
          disabled={disabled}
          onChange={this.internalOnChange}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
        /&gt;
        {this.renderButton()}
        {this.renderErrorText()}
      &lt;/div&gt;
    );
  }
}
&nbsp;
Input.propTypes = {
  /** Input Placeholder/Label */
  placeholder: PropTypes.string,
  /** Input field type: text, number, password and etc */
  type: PropTypes.string,
  /** Callback function for event */
  onChange: PropTypes.func.isRequired,
  /** Add star for label if this field is required */
  isRequired: PropTypes.bool,
  /** Error indication */
  hasError: PropTypes.bool,
  /** Error message */
  errorText: PropTypes.string,
  /** Disabled input */
  disabled: PropTypes.bool,
};
&nbsp;
Input.defaultProps = {
  placeholder: undefined,
  type: 'text',
  isRequired: false,
  hasError: false,
  errorText: undefined,
  disabled: false,
};
&nbsp;
export default Input;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Sep 14 2018 16:36:00 GMT+0300 (EEST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
