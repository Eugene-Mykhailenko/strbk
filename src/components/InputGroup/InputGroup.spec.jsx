import React from 'react';
import InputGroup from './InputGroup';
import Input from '../Input/Input';

jest.mock('../Icon/importIcons');

describe('InputGroup', () => {
  it('should render children', () => {
    const inputGroup = shallow(
      <InputGroup>
        <Input onChange={() => {}} />
      </InputGroup>,
    );

    expect(inputGroup.children().length).toEqual(1);
    expect(inputGroup.children().props().type).toEqual('text');
    expect(inputGroup).toMatchSnapshot();
  });
});
