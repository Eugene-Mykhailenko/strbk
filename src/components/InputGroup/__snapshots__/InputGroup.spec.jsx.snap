// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render children 1`] = `
<div
  className="input-group"
>
  <Input
    disabled={false}
    hasError={false}
    isRequired={false}
    onChange={[Function]}
    type="text"
  />
</div>
`;
